src/distilabel/utils/requirements.py

```diff
@@ -0,0 +1,57 @@
+# Copyright 2023-present, Argilla, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import TYPE_CHECKING, List, Union
+
+if TYPE_CHECKING:
+    from distilabel.steps import Step
+
+
+def add_requirements(requirements: Union[List[str]]) -> "Step":
```

Comment:
I would call this `requirements` instead of `add_requirements`, I think is more declarative.

Reply:
Also, the return type hint is wrong, it should be `Callable[["Step"], "Step"]`. Also, we should create a `TypeVar` bound to `_Step`. With the current setup the type of any class decorated will be coerced to `Step` (won't keep their original type)